test/distributed/test_c10d_common.py

```diff
@@ -1325,6 +1325,38 @@ def test_send_recv(self):
 
 instantiate_parametrized_tests(CommonDistributedDataParallelTest)
 
+class ProcessGroupWithDispatchedCollectivesTests(MultiProcessTestCase):
+    def setUp(self):
+        super(ProcessGroupWithDispatchedCollectivesTests, self).setUp()
+        self._spawn_processes()
+
+    def tearDown(self):
+        super(ProcessGroupWithDispatchedCollectivesTests, self).tearDown()
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    def _call_collective_with_varying_tensors(self, collective_func, *args):
+        # call collective with varying tensors to ensure that the tensors are
+        # correctly dispatched
+
+        # negative test to make sure a non-supported device fails during dispatch call
+        nonsupported_device = torch.device("meta")
```

Comment:
maybe I'm missing something but it doesn't see like there's tests with CPU and GPU tensors?

Reply:
Right, since this is a big change we are trying to do it piecewise. As of this PR, ProcessGroup doesn't support a list of backends, so we are still using ProcessGroupNCCL and ProcessGroupGloo to test this. Later once ProcessGroup adds that support, this test will also be updated to test the dispatching of CPU and GPU tensors